app/code/community/Bolt/Boltpay/Model/BoltOrder.php

```diff
@@ -593,6 +593,10 @@ public function correctBillingAddress(&$billingAddress, $fallbackAddress = null,
             $billingAddress->setCompany($fallbackAddress->getCompany());
         }
 
+        if ( $wasCorrected && !trim($billingAddress->getEmail())) {
```

Comment:
@johnmccombs1 
Probably, in the spirit of this, you wouldn't check for `$wasCorrected` as a condition, and only the email address not being empty.  Then you would set the `$wasCorrected` to true within the block.  Do you have a different thought?